core-contracts/src/bns-client.ts

```diff
@@ -0,0 +1,212 @@
+import { Client, Provider, Receipt, Result } from "@blockstack/clarity";
+import ripemd160 from 'ripemd160';
+import shajs from 'sha.js';
+
+export interface PriceFunction {
+  buckets: number[],
+  base: number,
+  coeff: number,
+  nonAlphaDiscount: number,
+  noVoyelDiscount: number,
+}
+
+export class BNSClient extends Client {
+  constructor(provider: Provider) {
+    super("S1G2081040G2081040G2081040G208105NK8PE5.bns", "bns", provider);
+  }
+
+  // (namespace-preorder (hashed-namespace (buff 20))
+  //                     (stx-to-burn uint))
+  async namespacePreorder(namespace: string, 
+                          salt: string,
+                          STX: number, 
+                          params: { sender: string }): Promise<Receipt> {
+    if (namespace === '') {
+      throw new Error("Namespace can't be empty");
+    }
+    if (STX <= 0) {
+      throw new Error("STX should be non-zero positive");
+    }
+
+    let sha256 = new shajs.sha256().update(`${namespace}${salt}`).digest();
+    let hash160 = new ripemd160().update(sha256).digest('hex');
+    let hashedNamespace = `0x${hash160}`;
+    const tx = this.createTransaction({
+      method: { name: "namespace-preorder", args: [`${hashedNamespace}`, `u${STX}`] }
+    });
+    await tx.sign(params.sender);
+    const res = await this.submitTransaction(tx);
+    return res;
+  }
+
+  // (namespace-reveal (namespace (buff 20))
+  //                   (namespace-version uint)
+  //                   (price-function (tuple (buckets (list 16 uint)) (base uint) (coeff uint) (nonalpha-discount uint) (no-voyel-discount uint)))
```

Comment:
Shouldn't `voyel` be `vowel`? (Comment applies to all instances of "voyel" in this code)

Reply:
Pardon my french :) - fixed with https://github.com/blockstack/stacks-blockchain/pull/2013/commits/0ccaee0cb30f5f1523c08a8cb6baa0fdb9037d70